src/handler/GenerateProjectHandler.ts

```diff
@@ -93,10 +105,20 @@ async function specifyTargetFolder(metadata: IProjectMetadata): Promise<vscode.U
     const OPTION_CHOOSE_ANOTHER_FOLDER: string = "Choose another folder";
     const LABEL_CHOOSE_FOLDER: string = "Generate into this folder";
     const MESSAGE_EXISTING_FOLDER: string = `A folder [${metadata.artifactId}] already exists in the selected folder. Continue to overwrite or Choose another folder?`;
+    const MESSAGE_FOLDER_NOT_EMPTY: string = "The selected folder is not empty. Existing files with same names will be overwritten. Continue to overwrite or Choose another folder?"
+    
+    const MESSAGE: string = metadata.createArtifactIdFolder ? MESSAGE_EXISTING_FOLDER : MESSAGE_FOLDER_NOT_EMPTY;
 
     let outputUri: vscode.Uri = metadata.defaults.targetFolder ? vscode.Uri.file(metadata.defaults.targetFolder) : await openDialogForFolder({ openLabel: LABEL_CHOOSE_FOLDER });
-    while (outputUri && await fse.pathExists(path.join(outputUri.fsPath, metadata.artifactId))) {
-        const overrideChoice: string = await vscode.window.showWarningMessage(MESSAGE_EXISTING_FOLDER, OPTION_CONTINUE, OPTION_CHOOSE_ANOTHER_FOLDER);
+    const projectLocation = metadata.createArtifactIdFolder ? path.join(outputUri.fsPath, metadata.artifactId) : outputUri.fsPath;

```

Comment:
I believe that would happen in the previous code as well, right? Not very experienced in TS.

Anyway, do you have a more specific recommendation for this? Or simply checking if the value is not undefined right after line 112 seems fine?

Reply:
I believe my changes now are more aligned with Virtual Workspaces compatibility. Instead of using `vscode.Uri.file()`, I've created them using `.parse()`, as the former adds the `scheme` to `path` as well, for some weird reason.

When showing the project path in `specifyOpenMethod()`, it'd be harder to read. Example:

```json
{
    "path": "/file:///Users/johndoe/projects/folder",
    "scheme": "file"
}
```

By using `vscode.Uri.parse("/Users/johndoe/projects/folder")`, this is the result:
```json
{
    "path": "/Users/johndoe/projects/folder",
    "scheme": "file"
}
```